osmoutils/slice_helper.go

```diff
@@ -49,3 +49,27 @@ func ContainsDuplicate[T any](arr []T) bool {
 	}
 	return false
 }
+
+type LessFunc[T any] func(a, b T) bool
+
+// MergeSlices efficiently merges two sorted slices into a single sorted slice.
+func MergeSlices[T any](slice1, slice2 []T, less LessFunc[T]) []T {
```

Comment:
// Functions are sorted using less func given, started from smallest going to greatest

Reply:
godoc improved